tests/rptest/tests/nodes_decommissioning_test.py

```diff
@@ -101,3 +109,121 @@ def node_removed():
 
         self.run_validation(enable_idempotence=False, consumer_timeout_sec=45)
+
+
+class SINodesDecommissioningTest(EndToEndTest):
+    """
+    Enable SI on a topic and test node decommissioning
+    """
+    segment_size = 1048576  # 1MB
+    s3_host_name = "minio-s3"
+    s3_access_key = "panda-user"
+    s3_secret_key = "panda-secret"
+    s3_region = "panda-region"
+    s3_topic_name = "panda-topic"
+    topics = []
+
+    def __init__(self, test_context):
+        super(SINodesDecommissioningTest,
+              self).__init__(test_context=test_context)
+
+        self.s3_bucket_name = f"panda-bucket-{uuid.uuid1()}"
+        self._extra_rp_conf = dict(
+            cloud_storage_enabled=True,
+            cloud_storage_enable_remote_read=True,
+            cloud_storage_enable_remote_write=True,
+            cloud_storage_access_key=self.s3_access_key,
+            cloud_storage_secret_key=self.s3_secret_key,
+            cloud_storage_region=self.s3_region,
+            cloud_storage_bucket=self.s3_bucket_name,
+            cloud_storage_disable_tls=True,
+            cloud_storage_api_endpoint=self.s3_host_name,
+            cloud_storage_api_endpoint_port=9000,
+            cloud_storage_reconciliation_interval_ms=500,
+            cloud_storage_max_connections=5,
+            log_segment_size=self.segment_size,  # 1MB
+            cloud_storage_cache_size=5 * self.segment_size,
+        )
+
+        self.scale = Scale(test_context)
+        self.redpanda = RedpandaService(
+            context=test_context,
+            num_brokers=4,
+            extra_rp_conf=self._extra_rp_conf,
+        )
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.s3_client = S3Client(
+            region=self.s3_region,
+            access_key=self.s3_access_key,
+            secret_key=self.s3_secret_key,
+            endpoint=f"http://{self.s3_host_name}:9000",
+            logger=self.logger,
+        )
+
+    def setUp(self):
+        self.s3_client.empty_bucket(self.s3_bucket_name)
+        self.s3_client.create_bucket(self.s3_bucket_name)
+        self.redpanda.start()
+
+        for partition_count in range(1, 5):
+            for replication_factor in (3, 3):
+                name = f'topic{len(self.topics)}'
+                spec = TopicSpec(name=name,
+                                 partition_count=partition_count,
+                                 replication_factor=replication_factor)
+                self.topics.append(spec)
+
+        for spec in self.topics:
+            self.kafka_tools.create_topic(spec)
+            self.topic = spec.name
+
+    def tearDown(self):
+        self.s3_client.empty_bucket(self.s3_bucket_name)
+
+    @cluster(num_nodes=6, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_decommissioning_working_node_with_si(self):
+        self.start_producer(1)
+        self.start_consumer(1)
+        self.await_startup()
+        admin = Admin(self.redpanda)
+
```

Comment:
> Should wait for some quantity of data to be produced (the consumer probably has an offset you can wait_until on?) so that the decom is definitely doing some real data movement.

The `await_startup` method does just that. It suspends execution until a minmum number of records are consumed.
```
  def await_startup(self, min_records=5, timeout_sec=30):
      try:
          wait_until(lambda: self.consumer.total_consumed() >= min_records,
                     timeout_sec=timeout_sec,
                     err_msg="Timed out after %ds while awaiting initial record delivery of %d records" %\
                     (timeout_sec, min_records))
      except BaseException:
          self._collect_all_logs()
          raise
```

> Maybe also wait until some of that data has made it into S3 so that we're truly exercising the SI part.
Good idea. I'll add it in the next push.

Reply:
Ah, I see - maybe also rename await_startup to await_records_consumed or something along those lines to make the test read more explicitly (for people like me who are  prone to misreading things ;-))